### PR TITLE
Create customised web context

### LINKED ISSUE #226: Adapt to construct properties for webkit
```
Commit: v0.9.0-21-g21890f2
GTK compile: 3.18.7
GTK run: 3.18.7
WebKit compile: 2.10.7
WebKit run: 2.10.7
WebKit2: 1
libsoup compile: 2.52.2
libsoup run: 2.52.2
```

Looks like some properties has been moved to construct only.

```
(uzbl-core:14457): GLib-GObject-WARNING **: g_object_set_valist: construct property "website-data-manager" for object 'WebKitWebContext' can't be set after construction
(uzbl-core:14457): GLib-GObject-WARNING **: g_object_set_valist: construct property "local-storage-directory" for object 'WebKitWebsiteDataManager' can't be set after construction
```

Similiar to `web-extension-dir` these should be command line flags and read-only variables.

### PR DESCRIPTION
**Pull request by keis**
Initialisation of website data manager is moved to main where the web
context is created. All read-only properties of data manager properly
marked as constant in variables.

Pass web context to gui for creating the web view.

Fixes #226

### CODE REVIEW

File: src/setup.h [hunk 1]
```diff
@@ -15,7 +17,7 @@ void
 uzbl_events_free ();
 
 void
-uzbl_gui_init ();
+uzbl_gui_init (WebKitWebContext *context);
```
Comment (keis) [inline src/setup.h]: Considering if it would be better to store the context on the main struct. Maybe nested under a UzblWeb struct? WDYT @mathstuf ?

Comment (mathstuf) [inline src/setup.h]: What thread will own the `UzblWeb` struct? Why not just put it in `UzblGui` or `UzblVariables`?

Comment (keis) [inline src/setup.h]: Yeah. gui would also work. The only difference would be that the webcontext is created a bit later (uzbl_gui_init) is the last to be called. I'll test that and see if it still works :)

File: src/variables.c [hunk 2]
```diff
@@ -1479,7 +1478,6 @@ uzbl_variables_private_new (GHashTable *table)
     }
 
 #if WEBKIT_CHECK_VERSION (2, 9, 4)
```
Comment (mathstuf) [inline src/variables.c]: This is now empty.

File: src/variables.c [hunk 3]
```diff
@@ -1409,23 +1408,23 @@ uzbl_variables_private_new (GHashTable *table)
         { "enable_pagecache",             UZBL_V_FUNC (enable_pagecache,                       INT)},
         { "enable_offline_app_cache",     UZBL_V_FUNC (enable_offline_app_cache,               INT)},
 #ifdef HAVE_LOCAL_STORAGE_PATH
-        { "local_storage_path",           UZBL_V_FUNC (local_storage_path,                     STR)},
+        { "local_storage_path",           UZBL_C_FUNC (local_storage_path,                     STR)},
```
Comment (mathstuf) [inline src/variables.c]: Should probably move all of these to the "constants" section.

File: README.md [hunk 4]
```diff
@@ -866,6 +850,23 @@ they are written as comments.
...
+  - Location to store cache files if no more specific directory is given.
+    Can be set at initialisation with `--cache-dir`
+* `base_data_directory` (string) (default: `$XDG_DATA_HOME/uzbl`) (WebKit2 >= 2.9.4)
+  - Location to store data files if no more specific directory is given.
+    Can be set at initialisation with `--data-dir`
+* `offline_app_cache_directory` (string) (default: `$XDG_CACHE_HOME/uzbl/applications`) (WebKit2 >= 2.9.4)
+  - Location to store files for offline applications.
+* `websql_directory` (string) (default: `$XDG_DATA_HOME/ubl/databases`) (WebKit2 >= 2.9.4)
+  - Location to store WebSQL databases.
+
```
Comment (mathstuf) [inline README.md]: This newline should go back (no left spacing where it came from).

File: src/variables.c (hunk 2 above)
Comment (keis) [inline src/variables.c]: Fixed

File: README.md (hunk 4 above)
Comment (keis) [inline README.md]: fixed

File: src/variables.c (hunk 3 above)
Comment (keis) [inline src/variables.c]: moved

**Comment by mathstuf** [comment]
+1

**Pull request merged**